pkg/proxy/server_test.go

```diff
@@ -73,7 +73,13 @@ func testServer(t *testing.T, scheme string, secure bool, delayTx bool) {
 		cfg.TLSInfo = tlsInfo
 	}
 	p := NewServer(cfg)
-	<-p.Ready()
+
+	select {
+	case <-p.Ready():
+	case err := <-p.Error():
+		t.Fatal(err)
+	}
```

Comment:
There are lots of duplicated code, can you wrap them into a common function or method?

Reply:
Done! There is a wrapper around these selects in `server_test.go` now. There is another possible approach:
A method on `(*server)`, but this requires a method definition on the `Server` interface as well. The method (tentative name `WaitUntilReady`) could either have a signature of:
`func () error`
And handling would be given as:
`if err := s.WaitUntilReady(); err != nil { [handle error] }`
or
`func (errcb func(error))`
with the handling as:
s.WaitUntilReady(func (err error) { [handle error] })`

Let me know if these approaches are more desirable.